src/coinbase/address/external_address.ts

```diff
@@ -217,33 +232,132 @@ export class ExternalAddress extends Address {
    */
   private async getStakingBalances(
     assetId: string,
-    options: CoinbaseExternalAddressStakeOptions,
+    options: StakingContextOptions = { mode: StakeOptionsMode.DEFAULT },
   ): Promise<{ [key: string]: Decimal }> {
-    const request = {
-      network_id: this.getNetworkId(),
-      asset_id: Asset.primaryDenomination(assetId),
-      address_id: this.getId(),
-      options: this.transformStakeOptions(options),
-    };
+    let typedContext: StakingContextContext;
```

Comment:
We should find a way to remove all references to protocol specific code here.

Reply:
Yes - I think you are right. I am trying now to pass options as is to backend and let the backend do the defaulting and validations. Does that sound right to you ?

Reply:
![thumbs-thumbs-up-kid](https://github.com/coinbase/coinbase-sdk-nodejs/assets/143219592/c10d910c-ad63-47e0-98db-dbbe4ca1e7c9)